core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -2127,21 +2118,15 @@ class KafkaApisTest {
       error = Errors.NONE
     ))
 
-    val response = readResponse(joinGroupRequest, capturedResponse)
-      .asInstanceOf[JoinGroupResponse]
+    val response = capturedResponse.getValue.asInstanceOf[JoinGroupResponse]
 
     assertEquals(Errors.NONE, response.error)
     assertEquals(0, response.data.members.size)
     assertEquals(memberId, response.data.memberId)
     assertEquals(0, response.data.generationId)
     assertEquals(memberId, response.data.leader)
     assertEquals(protocolName, response.data.protocolName)
-
-    if (version >= 7) {
```

Comment:
Why is this not needed?

Reply:
Now these test cases are working with the raw response object returned by `KafkaApis`. This means we don't have to deal with fields which are dropped during serialization.